backend/src/routes/api/notebooks/utils.ts

```diff
@@ -27,12 +29,24 @@ export const getNotebookStatus = async (
   const notebookName = notebook?.metadata.name;
   let newNotebook: Notebook;
   if (isRunning && !notebook?.metadata.annotations?.['opendatahub.io/link']) {
-    const route = await getRoute(fastify, namespace, notebookName).catch((e) => {
-      fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);
-      return undefined;
-    });
-    if (route) {
-      newNotebook = await patchNotebookRoute(fastify, route, namespace, notebookName).catch((e) => {
+    const disableServiceMesh = getDashboardConfig().spec.dashboardConfig.disableServiceMesh;
+    let host: string;
+    if (!disableServiceMesh) {
+      host = await getServiceMeshGwHost(fastify, namespace).catch((e) => {
+        fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);
+        return undefined;
+      });
+    } else {
+      const route = await getRoute(fastify, namespace, notebookName).catch((e) => {
+        fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);
+        return undefined;
+      });
+      if (route) {
+        host = route.spec.host;
+      }
+    }
+    if (host) {
```

Comment:
You can save the check above if you do something like:

```sugestion
if(route?.spec.host) {
    newNotebook = await patchNotebookRoute(fastify, route.spec.host, namespace, notebookName).catch((e) => {
```

Reply:
I think my changes in latest commit 93aa3ae improve the logic there, but I don't thinkk using  

```
if(route?.spec.host) {
    newNotebook = await patchNotebookRoute(fastify, route.spec.host, namespace, notebookName).catch((e) => {
```

would work in this case since in the if (SM) statement we fetch the host directly, and else (no SM) we fetch the host from the route struct. Am I misinterpreting your comment?

Reply:
Yes - marking as resolved, feel free to reopen if needed.